sklearn/metrics/tests/test_pairwise_distances_reduction.py

```diff
@@ -1148,10 +1148,15 @@ def test_sqeuclidean_row_norms(
     spread = 100
     X = rng.rand(n_samples, n_features).astype(dtype) * spread
 
+    X_csr = csr_matrix(X)
+
     sq_row_norm_reference = np.linalg.norm(X, axis=1) ** 2
     sq_row_norm = np.asarray(sqeuclidean_row_norms(X, num_threads=num_threads))
 
+    sq_row_norm_csr = np.asarray(sqeuclidean_row_norms(X_csr, num_threads=num_threads))
```

Comment:
Not related to this PR changes: If `np.asrray` is needed here, it's because `sqeuclidean_row_norms` returns memoryviews which it mustn't . Hence I think we can move `np.asarray` inside `sqeuclidean_row_norms` as part of this PR.

Reply:
Yep I was about to ask why this function returns memoryviews, thanks